mlir/include/mlir/Dialect/GPU/IR/CompilationInterfaces.h

```diff
@@ -100,6 +106,10 @@ class TargetOptions {
   /// Compilation process target representation.
   CompilationTarget compilationTarget;
 
+  /// Parent symbol table of all the GPU modules being serialized. By default
+  /// this member is null as it is not required by most targets.
+  SymbolTable *parentTable;
```

Comment:
Can we make this a callback that would lazy initialize it on demand if the target wants it?
Then we don't need a pass option and we can remove this weird coupling where initializing the pass requires to know if the target wants a SymbolTable.

Reply:
I've switched it.